functions/New-DbaAgentJob.ps1

```diff
@@ -0,0 +1,476 @@
+            return
+        }
+
+        # Check of the event log level is of type string and set the integer value
+        if(($EmailLevel.GetType().Name -eq 'String') -and ($EventLogLevel.Length -ge 1))
+        {
+            switch ($EventLogLevel)
+            {
+                {($_ -eq "Never") -or ($_ -eq '')} { $EventLogLevel = 0 } 
+                "OnSuccess" { $EventLogLevel = 1 } 
+                "OnFailure" { $EventLogLevel = 2 }
+                "Always" { $EventLogLevel = 3 } 
+            }
+        }
+
+        # Check of the email level is of type string and set the integer value
+        if(($EmailLevel.GetType().Name -eq 'String') -and ($EmailLevel.Length -ge 1))
+        {
+            switch ($EmailLevel)
+            {
+                {($_ -eq "Never") -or ($_ -eq '')} { $EmailLevel = 0 } 
+                "OnSuccess" { $EmailLevel = 1 } 
+                "OnFailure" { $EmailLevel = 2 }
+                "Always" { $EmailLevel = 3 } 
+            }
+        }
+
+        # Check of the net send level is of type string and set the integer value
+        if(($NetsendLevel.GetType().Name -eq 'String') -and ($NetsendLevel.Length -ge 1))
+        {
+            switch ($NetsendLevel)
+            {
+                {($_ -eq "Never") -or ($_ -eq '')} { $NetsendLevel = 0 } 
+                "OnSuccess" { $NetsendLevel = 1 } 
+                "OnFailure" { $NetsendLevel = 2 }
+                "Always" { $NetsendLevel = 3 } 
+            }
+        }
+
+        # Check of the page level is of type string and set the integer value
+        if(($PageLevel.GetType().Name -eq 'String') -and ($PageLevel.Length -ge 1))
+        {
+            switch ($PageLevel)
+            {
+                {($_ -eq "Never") -or ($_ -eq '')} { $PageLevel = 0 } 
+                "OnSuccess" { $PageLevel = 1 } 
+                "OnFailure" { $PageLevel = 2 }
+                "Always" { $PageLevel = 3 } 
+            }
+        }
+
+        # Check of the delete level is of type string and set the integer value
+        if(($DeleteLevel.GetType().Name -eq 'String') -and ($DeleteLevel.Length -ge 1))
+        {
+            switch ($DeleteLevel)
+            {
+                "Never" { $DeleteLevel = 0 } 
+                "OnSuccess" { $DeleteLevel = 1 } 
+                "OnFailure" { $DeleteLevel = 2 }
+                "Always" { $DeleteLevel = 3 } 
+            }
+        }
+
+        [object[]]$JobsCreated = $null
+    }
+
+    PROCESS
+    {
+        # Try connecting to the instance
+        Write-Message -Message "Attempting to connect to Sql Server.." -Level 2 -Silent $Silent
+        try 
+        {
+            $Server = Connect-SqlServer -SqlServer $SqlServer -SqlCredential $SqlCredential
+        }
+        catch 
+        {
+            Stop-Function -Message "Could not connect to Sql Server instance" -Silent $Silent -Target $SqlServer 
```

Comment:
`Stop-Function -Message "Could not connect to Sql Server instance" -Silent $Silent -Target $SqlServer`
becomes
`Stop-Function -Message "[$SqlServer] Could not connect to Sql Server instance: $($_.Exception.Message)" -Silent $Silent -Target $SqlServer -InnerErrorRecord $_ -Continue`
(The `-Continue ` is a requirement when the loop you asked for comes)